### PR TITLE
[SDESK-7062] Allow registering new resources for autocomplete suggestions

### PR DESCRIPTION
**Pull request by MarkLark86**

### CODE REVIEW

File: features/archive_autocomplete.feature [hunk 1]
```diff
@@ -34,3 +34,14 @@ Feature: Archive autocomplete
             {"value": "PUBLISHED-C"}
         ]}
         """
+        When we get "/archive_autocomplete?field=slugline&language=en&resources=archive"
```
Comment (petrjasek) [inline features/archive_autocomplete.feature:37]: I'm not sure if we should add that resources param, afaik there should be always suggestions from all configured resources

File: features/archive_autocomplete.feature [hunk 2]
```diff
@@ -34,3 +34,14 @@ Feature: Archive autocomplete
...
+        Then we get list with 3 items
+        """
+        {"_items": [
+            {"value": "PUBLISHED-A"},
+            {"value": "PUBLISHED-B"},
+            {"value": "PUBLISHED-C"}
```
Comment (petrjasek) [inline features/archive_autocomplete.feature:43]: might be useful to add the document count to the payload so we could sort it also based on that

File: features/archive_autocomplete.feature (hunk 1 above)
Comment (MarkLark86) [reply features/archive_autocomplete.feature:37]: I was just forward thinking there. I will remove it if you don't foresee any use cases

Comment (petrjasek) [reply features/archive_autocomplete.feature:37]: It might be useful to limit it, but imo would be also useful to get them all together by default

Comment (MarkLark86) [reply features/archive_autocomplete.feature:37]: It does that now. If no argument is provided, it uses all registered resources

File: apps/archive/autocomplete.py [hunk 3]
```diff
@@ -16,62 +16,89 @@
...
-        query = {
-            "query": {
-                "bool": {
-                    "filter": [
-                        {"term": {"state": "published"}},
-                        {"term": {"language": language}},
-                        {"range": {"versioncreated": {"gte": versioncreated_min}}},
-                    ],
-                },
+        return ListCursor([{"value": key, "count": all_suggestions[key]} for key in sorted(all_suggestions.keys())])
```
Comment (MarkLark86) [inline apps/archive/autocomplete.py:58]: @petrjasek I kept the sorting based on bucket key to keep with same functionality.

If we need to, we can modify the sort from the front-end (such as hoisting those that have `count > 1` to the top, and leave the rest sorted as is).

**Review by petrjasek — APPROVED** [review]

**Pull request merged**